### PR TITLE
CAMS-741 Improve district/division dropdown display and add auto-upgrade to All

### PR DESCRIPTION
**Pull request by mattstankey**
# Bug

When a division was selected in the District (Division) dropdown, the input field showed only the division name (e.g., "Juneau") instead of the full district and division (e.g., "District of Alaska (Juneau)"), making it inconsistent with the pill display.

# Purpose

Improve clarity in the trustee filter by showing both district and division name when a selection is made, and automatically upgrading to the district-level "All" option when a user has individually selected every division in a district.

# Major Changes

* Updated `getDistrictDivisionComboOptions` in `court-utils.ts` to use the full `"District (Division)"` format for `selectedLabel` (was division name only)
* Added `autoUpgradeToAll` function in `trusteeDistrictFilterUseCase.ts` — when all individual divisions for a district are selected, the selection is automatically replaced with the `|ALL` form
* Wired `autoUpgradeToAll` into `handleFilterCombined` so the upgrade fires on every user selection change
* Added `stryker` binary and `@stryker-mutator/core` package to knip ignore lists to suppress false-positive warnings
* Fixed `vitest/no-conditional-expect` lint violations in `sanitization.test.ts` inherited from main

# Testing/Validation

Implemented using TDD. Unit tests added for `autoUpgradeToAll` covering: partial selection (no upgrade), full selection (upgrades to ALL), multi-district scenarios, already-ALL selected, and empty inputs. Integration test added to `handleFilterCombined` verifying the full pipeline stores the `|ALL` form when all divisions are selected. Existing 54 tests in the use case file continue to pass.

Post-implementation spec review was performed and findings addressed: moved `vi.restoreAllMocks()` to `beforeEach`, replaced `vi.mock` with `vi.spyOn` for `UseApplicationInsights`, removed duplicate mutual-exclusion tests that re-tested `resolveCombinedSelections` through the use-case wrapper.

# Notes

`autoUpgradeToAll` runs after `resolveCombinedSelections` in `handleFilterCombined`. This ordering matters: `resolveCombinedSelections` handles the mutual-exclusion (selecting ALL removes specifics, selecting a specific removes ALL), then `autoUpgradeToAll` handles the implicit upgrade when all specifics are checked individually.

The `selectedLabel` fix is a one-line change in `court-utils.ts` — the full label was already being set as the dropdown option `label`, just not carried through to `selectedLabel` which drives what the combobox input displays after selection.

# Definition of Done:

- [ ] Code refactored for clarity: Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed: More important code doesn't directly depend on less important code
- [ ] Development debt eliminated: UX and code aligns to the team's latest understanding of the domain
- [ ] No regressions: Changes do not cause regression in related or unrelated areas of the application

## Summary by Sourcery

Improve trustee district/division selection UX and keep division selections consistent by auto-upgrading full-division selections to district-level "All" options.

Enhancements:
- Show full "District (Division)" text in the district/division dropdown input for selected divisions.
- Automatically replace fully selected sets of divisions with their corresponding district-level "All" option when handling combined trustee filters.

Tests:
- Add unit tests for the auto-upgrade-to-All behavior across partial, full, multi-district, already-All, and empty selection scenarios.
- Add an integration test for the combined filter handler to verify that fully selected divisions are stored as the district-level All option.
- Simplify sanitization error tests to use single regex-based expectations and resolve conditional-expect lint violations.

Chores:
- Refactor application insights mocking in trustee district filter tests to use spies and shared setup.

### CODE REVIEW

File: user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts [hunk 1]
```diff
@@ -16,6 +16,52 @@ import {
...
+  for (const [courtId, selectedCodes] of selectedSpecificByDistrict.entries()) {
+    const allCodes = allDivisionsByDistrict.get(courtId);
+    if (
+      allCodes &&
+      selectedCodes.size === allCodes.size &&
+      [...selectedCodes].every((c) => allCodes.has(c))
+    ) {
+      result = result.filter((s) => {
+        const [sCourt, sCode] = s.value.split('|');
+        return !(sCourt === courtId && sCode !== 'ALL');
```
Comment (cruzjone-flexion) [inline user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts]: Minor Suggestion: reverse the logi

`sCourt !== courtId || sCode === 'ALL'`

File: user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts [hunk 2]
```diff
@@ -16,6 +16,52 @@ import {
 } from '@/lib/utils/court-utils';
 import { AppointmentChapterType, formatChapterType } from '@common/cams/trustees';
 
+export function autoUpgradeToAll(
+  selections: ComboOption[],
+  districts: CourtDivisionDetails[],
+): ComboOption[] {
+  const allDivisionsByDistrict = new Map<string, Set<string>>();
```
Comment (cruzjone-flexion) [inline user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts:23]: Is it worth constructing this set once outside the method that runs after each filter modification?

**Review by cruzjone-flexion — APPROVED** [review]
Everything looks good to me, added some suggestions but not a deal breaker;

File: user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts (hunk 1 above)
Comment (mattstankey) [reply user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts]: ✅ Done — updated to `sCourt !== courtId || sCode === 'ALL'`.

File: user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts (hunk 2 above)
Comment (mattstankey) [reply user-interface/src/trustees/filters/trusteeDistrictFilterUseCase.ts:23]: ✅ Done — added a `courtNameById` map built in the same loop as `allDivisionsByDistrict`, so both are precomputed once from `districts` before any selection work. Replaced the `districts.find(...)` call at line 54 with a `courtNameById.get(courtId)` lookup.

**Review by jalba-flexion — APPROVED** [review]

**Review by diabagatekelly — APPROVED** [review]
